CHANGELOG.md

```diff
@@ -119,6 +119,9 @@ Unreleased changes template.
   change. Fixes [#2468](https://github.com/bazelbuild/rules_python/issues/2468).
 + (gazelle) Gazelle no longer ignores `setup.py` files by default. To restore
   this behavior, apply the `# gazelle:python_ignore_files setup.py` directive.
+* Don't re-fetch whl_library, python_repository, etc. repository rules
```

Comment:
I'm not exactly sure what to put in the parentheses here. This change affects anything that transitively calls which_unchecked, which I think includes whl_library (the failure mode I observed), python_repository, and local_runtime_repo. Feel free to re-word.

Reply:
LGTM